src/lua/mod.rs

```diff
@@ -218,61 +202,173 @@ fn thread_handle_message(sender: &Sender<LuaResponse>,
                 if let Err(err) = result {
                     warn!("thread: Error executing {}!", name);
 
-                    thread_send(&sender, LuaResponse::Error(err));
+                    thread_send(request.reply, LuaResponse::Error(err));
                 }
                 else {
                     trace!("thread: Execution of {} successful.", name);
+                    thread_send(request.reply, LuaResponse::Pong);
                 }
             }
             else { // Could not open file
                 // Unwrap_err is used because we're in the else of let
                 let read_error =
                     LuaError::ReadError(try_file.unwrap_err());
 
-                thread_send(&sender, LuaResponse::Error(read_error));
+                thread_send(request.reply, LuaResponse::Error(read_error));
             }
         },
 
-        LuaQuery::GetVariable(varname) => {
-            trace!("thread: Received request to get variable {}", varname);
-            let var_result = lua.get(varname.as_str());
+        LuaQuery::GetValue(varname) => {
+            trace!("thread: Received request to get variable {:?}", varname);
 
-            match var_result {
-                Some(var) => {
-                    thread_send(&sender, LuaResponse::Variable(Some(var)));
+            if varname.len() == 0 {
+                thread_send(request.reply, LuaResponse::InvalidName);
+                return;
+            }
+            // Table[0] String had to be cloned, it'd be nice if Rust let us
+            // borrow out parts of memory
+            match lua.get::<AnyLuaValue, _>(varname[0].clone()) {
```

Comment:
If you want to borrow out the part of the memory, why can't you used `varname[0].borrow()`?